eo-phi-normalizer/src/Language/EO/Phi/Dataize.hs

```diff
@@ -50,14 +52,20 @@ dataizeStepChain obj@(Formation bs)
       return (ctx, Right bytes)
   | Just (LambdaBinding (Function funcName)) <- listToMaybe [b | b@(LambdaBinding _) <- bs]
   , not hasEmpty = do
-      logStep ("Evaluating lambda '" <> funcName <> "'") (Left obj)
-      msplit (evaluateBuiltinFunChain funcName obj ()) >>= \case
-        Nothing -> do
-          ctx <- getContext
-          return (ctx, Left obj)
-        Just ((obj', _state), _alts) -> do
-          ctx <- getContext
-          return (ctx, Left obj')
+      ctx' <- getContext
+      if HashSet.member funcName knownAtomNames && not (HashSet.member funcName ctx'.enabledAtomNames)
```

Comment:
I would factor this condition into a separate function.

Reply:
Addressed in https://github.com/objectionary/normalizer/pull/426/commits/32be6c98f22e44833723d70e3c27f0607e56cdd5

However, I'm not sure I understood your suggestion correctly.